storage/innobase/srv/srv0start.cc

```diff
@@ -3936,21 +3944,63 @@ static lsn_t srv_shutdown_log() {
 
   srv_wake_log_tracker_thread();
 
-  if (srv_downgrade_logs) {
+  /* if upgrade is failed, delete and create new empty redo so old version do
+   not complaint  */
+  if (dd_init_failed_during_upgrade) {
+    ib::info(ER_IB_MSG_DOWNGRADING_LOG_FILE);
+
+    ut_ad(fil_space_t::s_redo_space != nullptr);
```

Comment:
Please, @rahulmalik87 , could we extract this functionality in a independent function (i.e.`recreate_redo_log_files)?

This way, `srv_shutdown_log` is not so long and the functionality is easier to maintain.